app/src/main/java/app/wefridge/wefridge/PantryFragment.kt

```diff
@@ -31,69 +34,59 @@ class PantryFragment : Fragment(), OnItemsChangeListener {
         return binding.root
     }
 
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        if (savedInstanceState?.getBoolean("getItemsSuccessfullyCalled") == true) {
-            setUpRecyclerViewWithItems()
-        } else {
-            val itemController = ItemController()
-            itemController.getItems({
-                    setUpRecyclerViewWithItems()
-                    getItemsSuccessfullyCalled = true
-                },
-                {
-                    displayAlertOnGetItemsFailed()
-                    getItemsSuccessfullyCalled = false
-                })
-        }
-
-        ItemController.addOnItemChangedListener(this)
+        val recyclerView = binding.list
 
+        with(recyclerView) {
+            layoutManager = LinearLayoutManager(context)
+            adapter = recyclerViewAdapter
+        }
 
         binding.fab.setOnClickListener {
-           findNavController().navigate(R.id.action_from_list_to_edit)
+            findNavController().navigate(R.id.action_from_list_to_edit)
         }
     }
 
-    override fun onSaveInstanceState(outState: Bundle) {
-        super.onSaveInstanceState(outState)
-        outState.putSerializable("getItemsSuccessfullyCalled", getItemsSuccessfullyCalled)
-    }
+    override fun onStart() {
+        super.onStart()
 
+        UserController.getCurrentUser() ?: return
 
-    override fun onDestroy() {
-        super.onDestroy()
-        ItemController.deleteOnItemChangedListener(this)
+        loadItems()
     }
 
-    private fun setUpRecyclerViewWithItems() {
-        val recyclerView = binding.list
-        with(recyclerView) {
-            layoutManager = LinearLayoutManager(context)
-            adapter = ItemRecyclerViewAdapter(ItemController.items, R.id.action_from_list_to_edit)
-        }
+    override fun onStop() {
+        super.onStop()
+        snapshotListener?.remove()
     }
 
+    private fun loadItems() {
+        snapshotListener?.remove()
+        with(values) {
+            val oldSize = size
+            clear()
+            recyclerViewAdapter.notifyItemRangeRemoved(0, oldSize)
+        }
 
-    private fun displayAlertOnGetItemsFailed() {
-        AlertDialog.Builder(requireContext())
-            .setTitle("Error loading your foodstuff")
-            .setMessage("Please check your internet connection and try again.")
-            .setPositiveButton("Retry") { _, _ ->
-                setUpRecyclerViewWithItems()
+        ItemController.getItemsSnapshot({
+            snapshotListener = it
+        }) { item, type, oldIndex, newIndex ->
+            when (type) {
```

Comment:
Can we get lost updates? As in one Person A modifies an item and Person B modifies it, so that only the modification of Person A are saved and what person B has wrote is lost?

Reply:
AFAIK this could be possible. To fix it, we can use [Firestore Transactions](https://firebase.google.com/docs/firestore/manage-data/transactions).